packages/@react-stately/color/src/useColorFieldState.ts

```diff
@@ -85,13 +85,12 @@ export function useColorFieldState(
     }
   };
 
-  let prevValue = useRef(colorValue);
-  if (prevValue.current !== colorValue) {
+  let [prevValue, setPrevValue] = useState(colorValue);
+  if (prevValue !== colorValue) {
     setInputValue(colorValue ? colorValue.toString('hex') : '');
-    prevValue.current = colorValue;
+    setPrevValue(colorValue);
```

Comment:
i think we have to be careful if this pattern becomes more prevalent because I don't think it works with our useControlledState when we would think that it should

the setState returned by useControlledState will call onChange as a direct result of setting this, leading to us calling an update to state in another component during render. I *think* that breaks a rule?
https://github.com/adobe/react-spectrum/blob/bb42ad9170590b9f94c56bbecf7815b6a6cb8a61/packages/%40react-stately/utils/src/useControlledState.ts#L73

I don't know yet if there is action needed for this PR, it occurred to me as I was almost done first my first pass, so this is a note to look out for this in my next pass or to be told that I'm worrying about nothing :)

Reply:
True, though in the cases here I don't think any are useControlledState, they are all useState.